SharpSpades/Utils/Throw.cs

```diff
@@ -6,22 +6,16 @@ namespace SharpSpades.Utils
 {
     internal static class Throw
     {
-        public static void IfNull<T>(T? obj) where T : class
+        public static void If<T>(T? obj, Predicate<T?> predicate, Exception exception) //where T : class
         {
-            if (obj is null)
-                throw new ArgumentNullException();
+            if (predicate.Invoke(obj))
+                throw exception;
         }
```

Comment:
I think it's a bit redundant having a method that throws a given exception if a given condition isn't met as it's what regular if-statements do and the code doesn't really become much cleaner with it.

Also having to initialize a new exception object every time you want to use the helper method is propably not very good.